### ISSUE TITLE
End of Support for AWS SDK for JavaScript v2 effective September 8, 2025

### ISSUE THREAD

**Issue by Jeongyong-park** [OpenDroneMap/ClusterODM#122]
****************************************
PLEASE REMOVE THIS NOTE AFTER READING IT!

First of all, thank you for taking the time to report an issue.

Before you continue, make sure you are in the right place. Please open an issue only to report faults and bugs. For installation problems, questions, server configuration problems and feature requests, please open a topic on http://community.opendronemap.org. 

Every issue needs to follow the format specified below. Do not remove the questions, simply answer each question by typing below it. If you don't know how to answer a question, write so.
****************************************

### What is the problem?

AWS SDK for JavaScript v2 will be end of support in Sep 8, 2025 (Not 2024)

### What should be the expected behavior?

Maybe no more can  not use AWS ASR Provider

### How can we reproduce this? (What steps did you do to trigger the problem? If applicable, please include multiple screenshots of the problem! Be detailed)

Should be change to new version AWS SDK

**Comment by spwoodcock** [comment]
The new package is `@aws-sdk/client-s3` and migration guide is here: https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/migrating.html

**Comment by FJEANNOT** [comment]
Any news on this ? Do you need someone to draft a PR ?

**Comment by pierotofy** [comment]
We'd love help with the upgrade and making sure the upgrade doesn't break anything.

**Comment by spwoodcock** [comment]
Would be amazing if you want to pick up https://github.com/OpenDroneMap/ClusterODM/pull/128 while I'm not able to right now @FJEANNOT 🙏 

I'll get back to it soonish, but had some other urgent requests come up 😅 

In that PR I upgraded to the latest `"aws-sdk": "^2.1692.0"`, as a start to the upgrade to use `@aws-sdk/client-s3`.

Next todo:
- Improve full e2e test.
- Add testing for the clustering /cloud capabilities.
- Upgrade libraries further where possible, including the AWS SDK.

**Comment by FJEANNOT** [comment]
I'm still running clusterodm and i seems to have no issues for now

**Comment by Jeongyong-park** [comment]
> I'm still running clusterodm and i seems to have no issues for now


Are you storing the data in the aws S3 bucket?

**Comment by FJEANNOT** [comment]
Yes, i've had a weird issue on task end (`Cannot upload file to S3: null (InvalidArgument)`), but i'm not sure this is related. I've fixed it by adding acl: "none" to the s3 section of the provider configuration.
